gateway-server/src/main/java/org/apache/knox/gateway/GatewayFilter.java

```diff
@@ -150,7 +153,13 @@ public void doFilter( ServletRequest servletRequest, ServletResponse servletResp
       }
     }
 
-    assignCorrelationRequestId();
+    /* If request contains X-Request-Id header use it else use random uuid as correlation id */
+    final String req_id = ((HttpServletRequest) servletRequest).getHeader(REQUEST_ID_HEADER_NAME);
```

Comment:
LGTM, just a nitpick here, why don't we use camel case names here?

Reply:
+1

Reply:
Yeah, I should, I'll update the PR, thanks for catching that!